src/JsonSchema/Constraints/BaseConstraint.php

```diff
@@ -146,12 +146,12 @@ public static function arrayToObjectRecursive($array)
         $json = json_encode($array);
         if (json_last_error() !== \JSON_ERROR_NONE) {
             $message = 'Unable to encode schema array as JSON';
-            if (version_compare(phpversion(), '5.5.0', '>=')) {
+            if (function_exists('json_last_error_msg')) {
                 $message .= ': ' . json_last_error_msg();
             }
             throw new InvalidArgumentException($message);
         }
 
-        return json_decode($json);
+        return (object) json_decode($json);
```

Comment:
Yes, it's deliberate. The JSON functions are already handling the deep cast - my understanding of this bug is that it's a shallow-cast problem only (that's the way it was reported).